include/zephyr/drivers/pinctrl/pinctrl_soc_kinetis_common.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022, 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * @file
+ * NXP Kinetis SOC specific helpers for pinctrl driver
+ */
+
+
+#ifndef ZEPHYR_SOC_ARM_NXP_KINETIS_COMMON_PINCTRL_SOC_H_
+#define ZEPHYR_SOC_ARM_NXP_KINETIS_COMMON_PINCTRL_SOC_H_
+
+/** @cond INTERNAL_HIDDEN */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+typedef uint32_t pinctrl_soc_pin_t;
+
+/* Kinetis KW/KL/KE series does not support open drain. Define macros to have no effect
+ * Note: KW22 and KW24 do support open drain, rest of KW series does not
+ */
+/* clang-format off */
+#if (defined(CONFIG_SOC_SERIES_KINETIS_KWX) &&                                                     \
+	!(defined(CONFIG_SOC_MKW24D5) || defined(CONFIG_SOC_MKW22D5))) ||                          \
+	defined(CONFIG_SOC_SERIES_KINETIS_KL2X) || defined(CONFIG_SOC_SERIES_KINETIS_KE1XF) ||     \
+	defined(CONFIG_SOC_SERIES_KE1XZ)
+#define PORT_PCR_ODE(x)   0x0
+#define PORT_PCR_ODE_MASK 0x0
+#endif
+/* clang-format on */
+
+/* Kinetis KE series does not support slew rate. Define macros to have no effect */
+#if defined(CONFIG_SOC_SERIES_KINETIS_KE1XF) || defined(CONFIG_SOC_SERIES_KE1XZ)
+#define PORT_PCR_SRE(x)   0x0
+#define PORT_PCR_SRE_MASK 0x0
+#endif
+
+#define Z_PINCTRL_KINETIS_PINCFG(node_id)                                                          \
```

Comment:
Beware that S32K1 is very similar (peripheral wise) to Kinetis KE1 series and both are using the same pinctrl driver. `Z_PINCTRL_KINETIS_PINCFG` is defined `zephyr/soc/nxp/s32/s32k1/pinctrl_soc.h`.

I don't think it's okay to move this header here, because even the hw IP is the same, each soc has an specific integration

Reply:
@manuargue do you have a recommendation on how this one should be handled? A more generic common name perhaps?

Reply:
I'm not requesting a change in the naming, although that's something we will need to do eventually but is a bigger change involving bindings, etc. My request on this particular change is to keep the pinctrl headers for Kinetis under `soc/nxp/kinetis/common/pinctrl_soc.h`. I think the whole point of the `pinctrl_soc.h` header is to be in the soc folder, hence the suffix `_soc` :)

Reply:
@manuargue completely disagree with this and this is a crucial part of the PR which is important for NPI we are trying to enable which will use this file that is not in the kinetis family, we don't want it in two places. This doesn't even affect s32 if it doesn't choose to include this header in it's pinctrl_soc.h,  kinetis still has it's own pinctrl_soc.h which is just including this file